### PR TITLE
replication: Avoid logging PreConditionFailed error

### PR DESCRIPTION
**Pull request by poornas**
## Description

ongoing replication and replication healing can be concurrent , giving rise to PreconditionFailed error. This can be ignored

## Motivation and Context
Avoid cluttering logs

## How to test this PR?
nothing to test.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/bucket-replication.go [hunk 1]
```diff
@@ -1359,12 +1359,16 @@ func (ri ReplicateObjectInfo) replicateAll(ctx context.Context, objectAPI Object
 			if err := replicateObjectWithMultipart(ctx, c, tgt.Bucket, object,
 				r, objInfo, putOpts); err != nil {
 				rinfo.ReplicationStatus = replication.Failed
```
Comment (harshavardhana) [inline cmd/bucket-replication.go]: why is replicationStatus failed for a good case @poornas ? isn't PrecondtionFailed a good scenario i.e the object is already replicated?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/bucket-replication.go (hunk 1 above)
Comment (poornas) [reply cmd/bucket-replication.go]: updated PR - we could just mark it completed if PreConditionFailed

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**